driver-core/src/main/java/com/datastax/driver/core/HostConnectionPool.java

```diff
@@ -115,7 +115,7 @@ public void run() {
 
         int toCreate = coreSize;
 
-        if (reusedConnection != null && reusedConnection.setOwner(this)) {
+        if (reusedConnection != null && toCreate > 0 && reusedConnection.setOwner(this)) {
```

Comment:
I was wondering if the connection was getting closed if we don't enter this condition; it's handled in the `finally` block in `Cluster.Manager.onUp` so 👍 .